x/farming/types/plan.go

```diff
@@ -249,17 +249,34 @@ type PlanI interface {
 	Validate() error
 }
 
-// ValidateRatioPlans validates a farmer's total epoch ratio and plan name.
-// A total epoch ratio cannot be higher than 1 and plan name must not be duplicate.
-func ValidateRatioPlans(i interface{}) error {
+// ValidateName validates duplicate plan name value.
+func ValidateName(i interface{}) error {
 	plans, ok := i.([]PlanI)
 	if !ok {
 		return sdkerrors.Wrapf(ErrInvalidPlanType, "invalid plan type %T", i)
 	}
 
-	totalEpochRatio := make(map[string]sdk.Dec)
 	names := make(map[string]bool)
```

Comment:
We can use `map[string]struct{}`(equivalent to a set) instead of `map[string]bool` for this kind of checking:

```
set := map[string]struct{}{}

set["foo"] = struct{}{}

if _, ok := set["foo"]; ok {
        // do something
}
```

`map[string]bool` takes up more space than `map[string]struct{}`.